dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt

```diff
@@ -53,58 +64,141 @@ class DataManager(
     }
 
     /**
-     * Method to make the data manager add data to a data store and store meta data in Dataland
+     * Method to make the data manager add data to a data store, store metadata in Dataland and sending messages to the
+     * relevant message queues
      * @param storableDataSet contains all the inputs needed by Dataland
      * @return ID of the newly stored data in the data store
      */
     @Transactional
-    fun addDataSet(storableDataSet: StorableDataSet, correlationId: String): String {
+    fun addDataSetToTemporaryStorageAndSendRequestQAMessage(storableDataSet: StorableDataSet, correlationId: String):
+        String {
         val company = companyManager.getCompanyById(storableDataSet.companyId)
         logger.info(
             "Sending StorableDataSet of type ${storableDataSet.dataType} for company ID " +
                 "${storableDataSet.companyId}, Company Name ${company.companyName} to storage Interface. " +
                 "Correlation ID: $correlationId",
         )
-        val dataId: String = storeDataSet(storableDataSet, company.companyName, correlationId)
-        metaDataManager.storeDataMetaInformation(
-            dataId,
-            storableDataSet.dataType,
-            storableDataSet.uploaderUserId,
-            storableDataSet.uploadTime,
-            company,
+        val dataId: String = storeDataSetInTemporaryStoreAndSendDataReceivedMessage(
+            storableDataSet,
+            company.companyName, correlationId,
+        )
+        val metaData = DataMetaInformationEntity(
+            dataId, storableDataSet.dataType.toString(),
+            storableDataSet.uploaderUserId, storableDataSet.uploadTime, company, QAStatus.Pending,
```

Comment:
I just discover: The backend sends two messages. That's not how we aligned it: 
The backend tells "qa received" - the internal storage reacts on that and sends "stored"
QA service reacts on "stored" and sends "qa-ed"
backend reacts on "qa-ed" and updates it's internal datamodel.

Reply:
This is now fixed with the refacotred message queue

Reply:
ok